### ISSUE TITLE
Add profile push limits

### ISSUE THREAD

**Issue by kolesnikovae** [grafana/pyroscope#2214]
The following tenant-level limits should be added in order to protect the write path from misuse/abuse:

1. `max_local_symbols_stacktraces_per_tenant` (~16M default)
2. `max_local_symbols_locations_per_tenant` (~4M default) (plus `functions` and `mappings`, optionally)
3. `max_local_symbols_strings_size_per_tenant` (~64MB default)

Optionally, each of them may have a **global** counterpart.

See https://github.com/grafana/pyroscope/pull/2157 for the context.

**Comment by simonswine** [comment]
Yes I think that sounds sensible. I do worry how this looks like from the user perspective and that they have often no options to easily counteract those exceeded limits.

All of those probably should contain bytes in their name and we should not forget to expose them separately using the reason label on `pyroscope_discarded_samples_total`

**Comment by kolesnikovae** [comment]
Good point regarding the user experience. The idea is that the limits should be high enough to only trigger if the input data is   malformed (intentionally or not). For example, if stack trace samples are generated randomly, or if the profiler produces "trash"
